// mindee/src/parsing/common/page.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Page = void 0;
const cropperExtra_1 = require("./extras/cropperExtra");
const extras_1 = require("./extras/extras");
const orientation_1 = require("./orientation");
class Page {
    constructor(predictionType, rawPrediction, pageId, orientation) {
        if (pageId !== undefined && orientation !== undefined) {
            this.orientation = new orientation_1.OrientationField({
                prediction: orientation,
                pageId: pageId,
            });
        }
        else {
            orientation = undefined;
        }
        this.id = pageId;
        this.prediction = new predictionType(rawPrediction["prediction"], pageId);
        if (rawPrediction["extras"] &&
            Object.keys(rawPrediction["extras"].length > 0)) {
            const extras = {};
            Object.entries(rawPrediction["extras"]).forEach(([extraKey, extraValue]) => {
                switch (extraKey) {
                    case "cropper":
                        extras["cropper"] = new cropperExtra_1.CropperExtra(extraValue);
                }
            });
            this.extras = new extras_1.Extras(extras);
        }
    }
    toString() {
        const title = `Page ${this.id}`;
        return `${title}
${"-".repeat(title.length)}
${this.prediction.toString()}
`;
    }
}
exports.Page = Page;
